compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -749,6 +749,29 @@ pub enum TerminatorKind<'tcx> {
     },
 }
 
+impl TerminatorKind<'_> {
```

Comment:
Can you put this in a different file? This one's mostly for type definitions

Reply:
There are similar functions in this file though, like `MirPhase::name`...

Not sure where to best put this

Reply:
Hmm, ok. We should probably move those out too, don't worry about it for this PR though

Reply:
The mir::terminator file looks like a proper place.